moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -1007,31 +1007,27 @@ bool MoveItConfigData::outputROSControllersYAML(const std::string& file_path)
 bool MoveItConfigData::output3DSensorPluginYAML(const std::string& file_path)
 {
   YAML::Emitter emitter;
-  emitter << YAML::BeginMap;
+  emitter << YAML::Comment("The name of this file shouldn't be changed, else the Setup Assistant won't detect it");
 
-  emitter << YAML::Comment("The name of this file shouldn't be changed, or else the Setup Assistant won't detect it");
+  emitter << YAML::BeginMap;
   emitter << YAML::Key << "sensors";
   emitter << YAML::Value << YAML::BeginSeq;
 
-  // Can we have more than one plugin config?
-  emitter << YAML::BeginMap;
-
-  // Make sure sensors_plugin_config_parameter_list_ is not empty
-  if (!sensors_plugin_config_parameter_list_.empty())
+  // Make sure sensors_plugin_config_parameter_list_ parameters map is not empty
+  if (!sensors_plugin_config_parameter_list_[0].empty())
```

Comment:
Can't `sensors_plugin_config_parameter_list` be empty?

Reply:
Right now I dont think it can be empty, it seems to always has the first plugin map
```
/// Sensor plugin configuration parameter list, each sensor plugin type is a map
std::vector<std::map<std::string, GenericParameter> > sensors_plugin_config_parameter_list_;
```

Reply:
That code snippet isn't convincing... i thought all vectors are initialized empty?

Reply:
@davetcoleman: yes, but not in this case. The code added in the initial PR does some unexpected things with vectors that makes `sensors_plugin_config_parameter_list_` always contain an element.

Reply:
Yes, that snippet doesn't really explain it all, so the reason  `sensors_plugin_config_parameter_list_` always has two elements is because we always fill it with the `default` sensor config on loading the setup assistant `moveit_setup_assistant/templates/moveit_config_pkg_template/config/sensors_3d.yml` 
```c++
bool MoveItConfigData::input3DSensorsYAML(const std::string& default_file_path, const std::string& file_path)

```
and when we write the user modified config we always put it as the first element in `sensors_plugin_config_parameter_list_` and write it.

Reply:
@MohmadAyman: why is there a "default sensor config"? It seems somewhat strange there is one, given that it's not necessarily true for a MoveIt configuration to use one.

I'd expect -- and I guess that is also what @davetcoleman expected -- the list to be empty if no sensor is configured.

That would avoid "strange" checks such as the one that sparked this discussion.